bpf/lib/process.h

```diff
@@ -409,6 +456,77 @@ static inline __attribute__((always_inline)) void execve_map_delete(__u32 pid)
 	}
 }
 
+// execve_joined_info_map_error() will increment the map error counter
+static inline __attribute__((always_inline)) void execve_joined_info_map_error(void)
+{
+	int one = MAP_STATS_ERROR;
+	__s64 *cntr;
+
+	cntr = map_lookup_elem(&tg_execve_joined_info_map_stats, &one);
+	if (cntr)
+		*cntr = *cntr + 1;
+}
+
+static inline __attribute__((always_inline)) int
+execve_joined_info_map_set(__u64 tid, struct execve_info *info)
+{
+	int err, zero = MAP_STATS_COUNT;
+	__s64 *cntr;
+
+	err = map_update_elem(&tg_execve_joined_info_map, &tid, info, BPF_ANY);
+	if (err < 0) {
+		/* -EBUSY or -ENOMEM with the help of the cntr error
+		 * on the stats map this can be a good indication of
+		 * long running workloads and if we have to make the
+		 * map size bigger for such cases.
+		 */
+		execve_joined_info_map_error();
+		return err;
+	}
+
+	cntr = map_lookup_elem(&tg_execve_joined_info_map_stats, &zero);
+	if (cntr && *cntr < EXECVE_JOINED_MAP_SIZE)
```

Comment:
Why are we checking against EXECVE_JOINED_MAP_SIZE here? What are we counting?

Reply:
ah left over when I was testing to fill up the map... btw forget I still didn't add the map sum accounting for metrics didn't find an example in the code base, so I will do it in separate PR